src/hooks/useAds.ts

```diff
@@ -0,0 +1,56 @@
+import { useQuery } from 'react-query';
+
+import { getMediaAds, getAdSchedule } from '#src/services/api.service';
+import { useConfigStore } from '#src/stores/ConfigStore';
+
+const CACHE_TIME = 60 * 1000 * 20;
+
+const useAdSchedule = ({ adScheduleId, enabled }: { adScheduleId: string | null | undefined; enabled: boolean }) => {
+  const { isLoading, data } = useQuery(
+    ['ad-schedule', adScheduleId],
+    async () => {
+      const adSchedule = await getAdSchedule(adScheduleId);
+
+      return adSchedule;
+    },
+    { enabled: enabled && !!adScheduleId, cacheTime: CACHE_TIME, staleTime: CACHE_TIME },
+  );
+
+  return {
+    isLoading,
+    data,
+  };
+};
+
+const useMediaAds = ({ jsonUrl, mediaId, enabled }: { jsonUrl: string | null | undefined; mediaId: string; enabled: boolean }) => {
+  const { isLoading, data } = useQuery(
+    ['media-ads', mediaId],
+    async () => {
+      // Waiting for `prd` deploy to remove `replace`
+      const mediaAds = await getMediaAds(jsonUrl?.replace('advertising/site', 'sites') as string, mediaId);
+
+      return mediaAds;
+    },
+    { enabled: enabled && !!mediaId, cacheTime: CACHE_TIME, staleTime: CACHE_TIME },
+  );
+
+  return {
+    isLoading,
+    data,
+  };
+};
+
+export const useAds = ({ mediaId }: { mediaId: string }) => {
+  const { adSchedule: adScheduleId, adScheduleUrls } = useConfigStore((s) => s.config);
```

Comment:
This confused me, because I forgot that we called the app config delivery property "AdScheduleUrls" and then if we have adScheudleUrls we disable the hook that is named "useAdSchedule". I think we can clear it up with just some renamings.

Maybe we should consider calling the perMedia use something more generic like "appBasedAds". That will allow it to make sense for customers who don't use per media, but do want to use our simplified ads instated of a stand alone ad schedule and also set us up to add more function to that system like ad frequency based on subscriptions for example without calling it all "perMediaAds".  Maybe the other hook can be something like useStandaloneAdSchedule?

Reply:
I agree. The naming is not perfect. Should we also deprecate useAdSchedule / useStandaloneAdSchedule hook? We hide this setting on the App Config page when Ad Config is active.

Reply:
We can call it `useLegacyStandaloneAds` or something. Unfortunately I think we'll need to leave it in for a while because otherwise we might break customers who have deployments with it.

Reply:
Fixed it